clang/test/CIR/CodeGen/vector-ext.cpp

```diff
@@ -213,3 +213,68 @@ void foo4() {
 // OGCG: %[[TMP2:.*]] = load i32, ptr %[[IDX]], align 4
 // OGCG: %[[ELE:.*]] = extractelement <4 x i32> %[[TMP1]], i32 %[[TMP2]]
 // OGCG: store i32 %[[ELE]], ptr %[[INIT]], align 4
+
+void foo9() {
```

Comment:
Can you add a test with a scalar shift value?

`vi4 splat_shl = a << 3;`

Also, a test with unsigned values and shift-right is needed.

Reply:
Both of them still require upstreaming `VecSplatOp`, I will upstream it next in another PR and add those test cases there, if that's okey